### ISSUE TITLE
`@ai-sdk/svelte` - `beforeUpdate(...)` cannot be used in runes mode

### ISSUE THREAD

**Issue by phenomen** [vercel/ai#3454]
### Description

After updating to Svelte 5, I get this error:
```
Uncaught Svelte error: lifecycle_legacy_only
`beforeUpdate(...)` cannot be used in runes mode
```
In the log it points to this code somewhere in `@ai-sdk/svelte`
![brave_zjXBUTPpHb](https://github.com/user-attachments/assets/659bb700-83d0-445d-9ecf-8d667bcb7d28)

### Code example

```svelte
<script lang="ts">
	import { useCompletion } from "@ai-sdk/svelte";

	const { input, handleSubmit, completion, isLoading } = useCompletion({
		api: "/api/openai/format",
	});

	let output = $state("");

	$effect(() => {
		output = $completion;
	});
</script>

```

**Comment by tguelcan** [comment]
Any workaround?

**Comment by chengwan7** [comment]
Encountered the same issue after upgrading to svelte 5.

**Comment by didier** [comment]
Yep this is fully broken in Svelte 5. Seems like a high priority issue to fix.

**Comment by phenomen** [comment]
Apparently, the code comes from [@ConsoleTVs/sswr](https://github.com/ConsoleTVs/sswr) which is [deprecated/unmaintained](https://github.com/ConsoleTVs/sswr/issues/52).

**Comment by vyconm** [comment]
I found multiple things to be broken when used directly in svelte 5 - my workaround so far has been to create a seperate svelte component with svelte 4 syntax, and load that in to the rest of my svelte 5 code. That way, the compiler will switch to legacy mode just for this part of code, while maintaining compatibility.

**Comment by himanshupht** [comment]
@vyconm Can you point me to an example?

**Comment by rawwerks** [comment]
hey vercel - doesn't the svelte creator work for you? please don't leave us all hanging.

**Comment by tguelcan** [comment]
> I found multiple things to be broken when used directly in svelte 5 - my workaround so far has been to create a seperate svelte component with svelte 4 syntax, and load that in to the rest of my svelte 5 code. That way, the compiler will switch to legacy mode just for this part of code, while maintaining compatibility.

As already mentioned here, the current workaround is:

`<svelte:options runes={false} />`

should be entered. But remember that unfortunately no runes can then be used in the same component.

**Comment by BrandonTing** [comment]
Hi 
any updates on the future roadmap of svelte support? Thanks

**Comment by TnTSpace** [comment]
> Any workaround?

The workaround I found was to step down to svelte legacy mode for the chat component, till vercel updates @ai-sdk/svelte

Meaning

For Props instead of:
- let { prop } = $props(), I used
- export let prop

For state Instead of:
- const loading = $state(false), I used
- $: loading = false

**Comment by BrandonTing** [comment]
`1.1.20` works for me. Thanks for your effort!

**Comment by lgrammel** [comment]
## Identify issue type

Identified as **Bug**  
Confidence: **High**

### Rationale
This is a concrete runtime compatibility failure in the existing Svelte SDK, later resolved by an updated package version.

### Evidence
- Using `useCompletion` from `@ai-sdk/svelte` in Svelte 5 runes mode throws `lifecycle_legacy_only` because dependency code calls `beforeUpdate(...)`.
- Multiple commenters reproduced the incompatibility and reported that switching the component to Svelte legacy mode was required as a workaround.
- A maintainer asked users to try `@ai-sdk/svelte@1.1.20`, and a reporter confirmed that version fixed the problem.
